core/src/com/unciv/logic/multiplayer/FriendList.kt

```diff
@@ -52,34 +52,25 @@ class FriendList {
         settings.save()
     }
 
-    fun getFriendsList(): MutableList<Friend> {
-        return friendList
-    }
+    fun getFriendsList() = friendList
 
     fun isFriendNameInFriendList(name: String): ErrorType {
-        for (index in friendList.indices) {
-            if (name == friendList[index].name) {
-                return ErrorType.ALREADYINLIST
-            }
+        return if (friendList.firstOrNull { it.name == name } != null ) {
```

Comment:
.. a call to the "new" getFriendByName would likely be inlined, thus giving the exact same result. Just FYI...